rand-core/src/lib.rs

```diff
@@ -0,0 +1,297 @@
+// Copyright 2017-2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// https://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Random number generation traits
+//! 
+//! This crate is mainly of interest to crates publishing implementations of
+//! `RngCore`. Other users are encouraged to use the
+//! [rand crate](https://crates.io/crates/rand) instead.
+//!
+//! `RngCore` is the core trait implemented by algorithmic pseudo-random number
+//! generators and external random-number sources.
+//! 
+//! `SeedableRng` is an extension trait for construction from fixed seeds and
+//! other random number generators.
+//! 
+//! `Error` is provided for error-handling. It is safe to use in `no_std`
+//! environments.
+//! 
+//! The `impls` and `le` sub-modules include a few small functions to assist
+//! implementation of `RngCore`. Since this module is only of interest to
+//! `RngCore` implementors, it is not re-exported from `rand`.
+
+#![doc(html_logo_url = "https://www.rust-lang.org/logos/rust-logo-128x128-blk.png",
+       html_favicon_url = "https://www.rust-lang.org/favicon.ico",
+       html_root_url = "https://docs.rs/rand-core/0.1")]
+
+#![deny(missing_debug_implementations)]
+
+#![cfg_attr(not(feature="std"), no_std)]
+#![cfg_attr(all(feature="alloc", not(feature="std")), feature(alloc))]
+
+#[cfg(feature="std")] extern crate core;
+#[cfg(all(feature = "alloc", not(feature="std")))] extern crate alloc;
+
+
+use core::default::Default;
+use core::convert::AsMut;
+
+#[cfg(all(feature="alloc", not(feature="std")))] use alloc::boxed::Box;
+
+pub use error::{ErrorKind, Error};
+
+
+mod error;
+pub mod impls;
+pub mod le;
+
+
+/// The core of a random number generator.
+/// 
+/// This trait encapsulates the low-level functionality common to all
+/// generators, and is the "back end", to be implemented by generators.
+/// End users should normally use [`Rng`] instead.
```

Comment:
This sounds like an old comment. Maybe point to the `rand` crate now, instead of only the `Rng` trait. (also in the rest of this comment block)